mne/epochs.py

```diff
@@ -818,9 +818,14 @@ def _reject_setup(self, reject, flat):
         # check for invalid values
         for rej, kind in zip((reject, flat), ("Rejection", "Flat")):
             for key, val in rej.items():
-                if val is None or val < 0:
+                if callable(val):
+                    continue
+                elif val is not None and val >= 0:
+                    continue
+                else:
                     raise ValueError(
-                        '%s value must be a number >= 0, not "%s"' % (kind, val)
+                        '%s value must be a number >= 0 or a valid function, not "%s"'
+                        % (kind, val)
                     )
 
         # now check to see if our rejection and flat are getting more
```

Comment:
One more thing, the `_reject_setup` function in epochs.py includes a section to "make sure new thresholds are at least as stringent as the old ones"